winch/codegen/src/codegen/call.rs

```diff
@@ -87,9 +125,9 @@ impl<'a> FnCall<'a> {
         masm.reserve_stack(self.total_arg_stack_space);
         self.assign_args(context, masm, <A as ABI>::scratch_reg());
         masm.call(callee);
-        masm.free_stack(self.spill_stack_space + self.total_arg_stack_space);
+        masm.free_stack(self.spill_stack_space + self.total_arg_stack_space + self.memory_consumed);
```

Comment:
I'm a little confused here: shouldn't we be popping just `total_arg_stack_space`, without also popping `memory_consumed` and `spill_stack_space`? It seems like this is double-counting, unless I'm misunderstanding what the above amounts mean.

(Labeling portions of the stack in the figure might help clarify, too, or else noting the state of hte stack at points in the sequence above)

Reply:
Agreed, I've added a figure of how the stack looks like at each point in time in https://github.com/bytecodealliance/wasmtime/commit/2e5372c8ffb430e75442ef33cc5cd7a3b408800e, hopefully this helps clarify!